mailjet_client_test.go

```diff
@@ -432,55 +454,188 @@ func TestUnitSendMail(t *testing.T) {
-		data := mailjet.ResultsV31{
-			ResultsV31: []mailjet.ResultV31{
-				{
-					To: []mailjet.GeneratedMessageV31{
-						{
-							Email:       "recipient@company.com",
-							MessageUUID: "ac93d194-1432-4e25-a215-2cb450d4a818",
-							MessageID:   87,
+	tests := []struct {
+		name           string
+		messages       mailjet.MessagesV31
+		mockResponse   interface{}
+		mockStatusCode int
+		wantResponse   *mailjet.ResultsV31
+		wantErr        interface{}
+	}{
+		{
+			name:     "sending successful",
+			messages: defaultMessages,
+			mockResponse: mailjet.ResultsV31{
+				ResultsV31: []mailjet.ResultV31{
+					{
+						To: []mailjet.GeneratedMessageV31{
+							{
+								Email:       "recipient@company.com",
+								MessageUUID: "ac93d194-1432-4e25-a215-2cb450d4a818",
+								MessageID:   87,
+							},
 						},
 					},
 				},
 			},
-		}
-		rawBytes, _ := json.Marshal(data)
-		return &http.Response{
-			Body:       ioutil.NopCloser(bytes.NewBuffer(rawBytes)),
-			StatusCode: http.StatusOK,
-		}, nil
-	}
-
-	m := mailjet.NewClient(httpClientMocked, mailjet.NewSMTPClientMock(true))
-
-	// We define parameters here to pass to SendMailV31
-	param := []mailjet.InfoMessagesV31{
-		mailjet.InfoMessagesV31{
-			From: &mailjet.RecipientV31{
-				Email: "passenger@mailjet.com",
-				Name:  "passenger",
+			mockStatusCode: 200,
+			wantResponse: &mailjet.ResultsV31{
+				ResultsV31: []mailjet.ResultV31{
+					{
+						To: []mailjet.GeneratedMessageV31{
+							{
+								Email:       "recipient@company.com",
+								MessageUUID: "ac93d194-1432-4e25-a215-2cb450d4a818",
+								MessageID:   87,
+							},
+						},
+					},
+				},
+			},
+			wantErr: nil,
+		},
+		{
+			name:     "authorization failed",
+			messages: defaultMessages,
+			mockResponse: mailjet.ErrorInfoV31{
+				Identifier: "ac93d194-1432-4e25-a215-2cb450d4a818",
+				StatusCode: 401,
+				Message:    "API key authentication/authorization failure. You may be unauthorized to access the API or your API key may be expired. Visit API keys management section to check your keys.",
+			},
+			mockStatusCode: 401,
+			wantResponse:   nil,
+			wantErr: &mailjet.ErrorInfoV31{
+				Identifier: "ac93d194-1432-4e25-a215-2cb450d4a818",
+				StatusCode: 401,
+				Message:    "API key authentication/authorization failure. You may be unauthorized to access the API or your API key may be expired. Visit API keys management section to check your keys.",
 			},
-			To: &mailjet.RecipientsV31{
-				mailjet.RecipientV31{
-					Email: "recipient@company.com",
+		},
+		{
+			name:     "simple errors in request",
+			messages: messagesWithAdvancedErrorHandling(),
+			mockResponse: mailjet.APIFeedbackErrorsV31{
+				Messages: []mailjet.APIFeedbackErrorV31{
+					{
+						Errors: []mailjet.APIErrorDetailsV31{
+							{
+								// TODO: Missing ErrorIdentifier in struct
+								// TODO: Missing ErorrCode in struct, but ErrorClass exists
+								//       -> Correct format not defined in API docs
```

Comment:
yeah, we should add those missing fields. Please add them

Reply:
I added them. Thanks!